src/cpp/src/generation_config.cpp

```diff
@@ -264,6 +268,25 @@ void GenerationConfig::validate() const {
     if (num_assistant_tokens == 0) {
         OPENVINO_ASSERT(max_ngram_size == 0, "'max_ngram_size' should be set to default value 0 when prompt lookup is disabled");
     }
+
+    if(is_structured_output_generation()) {
+        #ifndef ENABLE_XGRAMMAR
+            OPENVINO_THROW("Structured output is not supported in this build. Please, recompile with -DENABLE_XGRAMMAR=ON");
+        #else
+           (*structured_output_config).validate();
+        #endif
```

Comment:
We can check if there is any backend registered here or in the `validate` method.

Reply:
Done. Removed this last macros. Now we don't have any xgrammar macros in the sources.